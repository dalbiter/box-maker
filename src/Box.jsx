import React from 'react';

const Box = ({ width, height, color, idx, removeBox }) => {

    const handleClick = (e) => {
        const boxId = e.target.id;
        removeBox(boxId)   
    }
    return (
        <div>
            <br />
            <div className="Box" style={{width: `${width}px`, height: `${height}px`, backgroundColor: color}}>
                <button className="Box-remove-button" onClick={handleClick} id={idx}>X</button>
            </div>
        </div>
            
        
    )
};

export default Box;