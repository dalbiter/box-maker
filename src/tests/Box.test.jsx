import { render } from '@testing-library/react';
import Box from '../Box';
import { describe, expect, it } from 'vitest';

describe('Box', () => {
    it('renders Box component', () => {
        render(<Box />)
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<Box />)
        expect(asFragment()).toMatchSnapshot()
    })
});