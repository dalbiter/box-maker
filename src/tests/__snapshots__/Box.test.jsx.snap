// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box > should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <br />
    <div
      class="Box"
    >
      <button
        class="Box-remove-button"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
